website/www/site/content/en/get-started/quickstart/java.md

```diff
@@ -0,0 +1,185 @@
+  [JAVA_HOME](https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars001.html)
+  environment variable is set and points to your JDK installation.
+2. You can use either Gradle or Apache Maven to run this quickstart:
+    - Gradle: Install [Gradle](https://gradle.org/install/).
+    - Maven: Download [Apache Maven](https://maven.apache.org/download.cgi) and
+     follow the [installation guide](https://maven.apache.org/install.html)
+     for your operating system.
+
+## Clone the GitHub repository
+
+Clone or download the
+[apache/beam-starter-java](https://github.com/apache/beam-starter-java) GitHub
+repository and change into the beam-starter-java directory.
+
+{{< highlight >}}
+git clone https://github.com/apache/beam-starter-java.git
+cd beam-starter-java
+{{< /highlight >}}
+
+## Run the quickstart
+
+**Gradle**: To run the quickstart with Gradle, run the following command:
+
+{{< highlight >}}
+gradle run --args='--inputText=Greetings'
+{{< /highlight >}}
+
+**Maven**: To run the quickstart with Maven, run the following command:
+
+{{< highlight >}}
+mvn compile exec:java -Dexec.args=--inputText='Greetings'
+{{< /highlight >}}
+
+The output is similar to the following:
+
+{{< highlight >}}
+Hello
+World!
+Greetings
+{{< /highlight >}}
+
+The lines might appear in a different order.
+
+## Explore the code
+
+The main code file for this quickstart is **App.java** 
+([GitHub](https://github.com/apache/beam-starter-java/blob/main/src/main/java/com/example/App.java)).
+The code performs the following steps:
+
+1. Create a Beam pipeline.
+3. Create an initial `PCollection`.
+3. Apply a transform to the `PCollection`.
+4. Run the pipeline, using the Direct Runner.
+
+### Create a pipeline
+
+The code first creates a `Pipeline` object. The `Pipeline` object builds up the
+graph of transformations to be executed.
+
+{{< highlight >}}
+var options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+var pipeline = Pipeline.create(options);
+{{< /highlight >}}
+
+The `PipelineOptions` object lets you set various options for the pipeline. The
+`fromArgs` method shown in this example parses command-line arguments, which
+lets you set pipeline options through the command line.
+
+### Create an initial PCollection
+
+The `PCollection` abstraction represents a potentially distributed,
+multi-element data set. A Beam pipeline needs a source of data to populate an
+initial `PCollection`. The source can be bounded (with a known, fixed size) or
+unbounded (with unlimited size).
+
+This example uses the
+[`Create.of`](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/transforms/Create.html)
+method to create a `PCollection` from an in-memory array of strings. The
+resulting `PCollection` contains the strings "Hello", "World!", and a
+user-provided input string.
+
+{{< highlight >}}
+return pipeline
+	.apply("Create elements", Create.of(Arrays.asList("Hello", "World!", inputText)))
```

Comment:
Is this how @davidcavazos is spacing the example? I think in modern Java development we usually use 2 spaces for the first indent and 4 spaces for line wrapping: https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation

Reply:
Good catch, yeah let's stick to the Java style guide.

Reply:
done